### PR TITLE
Added support for any keyboard layout and some other keys

### PR DESCRIPTION
**Pull request by Microeinstein**
Keys are now handled with `GetKeyboardLayout()`, `GetKeyboardState()`, `ToAsciiEx()`
Added surrounding for angular braces `<>` and backtick `` ` ``

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,23 +1,21 @@
 # SurroundSelection
 
-[![Build status](https://ci.appveyor.com/api/projects/status/github/dail8859/SurroundSelection?branch=master&svg=true)](https://ci.appveyor.com/project/dail8859/SurroundSelection/branch/master)
```
Comment (dail8859) [inline README.md]: Keep this in

File: README.md [hunk 2]
```diff
@@ -1,23 +1,21 @@
...
+- `` ` ``
 - `(` or `)`
-- `{` or `}`
 - `[` or `]`
+- `{` or `}`
+- `<` or `>`
 
 ## Installation
-Install the plugin by downloading it from the [Release](https://github.com/dail8859/SurroundSelection/releases) page and copy `SurroundSelection.dll` to your `plugins` folder.
+Install the plugin by downloading it from the [Release](https://github.com/Microeinstein/SurroundSelection/releases) page and copy `SurroundSelection.dll` to your `plugins` folder.
```
Comment (dail8859) [inline README.md]: Keep the reference to `dail8859`

File: src/Main.cpp [hunk 3]
```diff
@@ -52,6 +57,14 @@ const wchar_t *GetIniFilePath() {
 	return iniPath;
 }
 
+static void updateKL() {
+	keyboardLayout = GetKeyboardLayout(0);
+}
+
+static WPARAM keyIfDown(UINT value) {
```
Comment (dail8859) [inline src/Main.cpp]: This isn't used anywhere.

File: src/Main.cpp [hunk 4]
```diff
@@ -117,49 +130,85 @@ static void SurroundSelectionsWith(char ch1, char ch2) {
...
-
-				if (ch1 != 0 && editor.GetSelectionEmpty() == 0) {
-					SurroundSelectionsWith(ch1, ch2);
-					return TRUE; // This key has been "handled" and won't propogate
-				}
-			}
+	if (ncode != HC_ACTION)
+		goto proceed;
+	bool keyState = (lparam >> 30) & 1;
+	/*if (isKey(wparam, VK_CONTROL) || isKey(wparam, VK_SHIFT) || isKey(wparam, VK_MENU)) {
```
Comment (dail8859) [inline src/Main.cpp]: All the commented out code doesn't need to stay.

**Review by dail8859 — CHANGES_REQUESTED** [review]
Thanks for the PR! This well definitely be great to support various keyboard layouts.

**Comment by Microeinstein** [comment]
Is everything okay with this latest commit?

**Comment by dail8859** [comment]
Looks good. Thanks again for the PR. Will merge it when I have a bit of free time.

**Review by dail8859 — APPROVED** [review]

**Pull request closed without merging**